### ISSUE TITLE
Add Short Name as admin settings field alongside Site Title

### ISSUE THREAD

**Issue by westonruter** [GoogleChromeLabs/pwa-wp#210]
The web app manifest removed the `short_name` because it was not reliable to automatically generate: https://github.com/xwp/pwa-wp/pull/201

In order to avoid a site having to add a `web_app_manifest` filter to supply the `short_name`, what if a new Short Name field is added after the Site Title:

![image](https://user-images.githubusercontent.com/134745/62659346-f2320580-b91f-11e9-91d2-e17b312b9061.png)

This field could be hidden until the Site Title field has more than 12 characters in it.

**Comment by iandunn** [comment]
🤔 

My first instinct is that it's not worth adding an extra option for this, but I do like your idea about hiding the field if the name is `< 12` characters.

I'd probably still lean towards requiring devs to use a filter or something though, or letting a plugin add that field, rather than putting it in Core though.

**Comment by AdelDima** [comment]
I agree with @iandunn, for me, I added a short name and name on Customizer settings, this way theme/plugin user can find all PWA options in one place for more simplicity.

![Screen Shot on 2019-12-03 at 15-40-13](https://user-images.githubusercontent.com/3769804/70060892-9cc2e800-15e3-11ea-9edd-85cf66664acc.png)

**Comment by westonruter** [comment]
Here's a prototype mini plugin which implements this Short Name UI on the general settings screen: https://gist.github.com/westonruter/dec7d190060732e29a09751ab99cc549

**Comment by westonruter** [comment]
This issue comes up over and over again in the support forums, so clearly there needs to be a UI to manage this. I'm adding it to the 0.7 milestone.

**Comment by pooja-muchandikar** [comment]
QA Passed ✅ 

- Short Name filed is available in General Settings
<img width="673" alt="Screenshot 2022-04-19 at 12 47 26 PM" src="https://user-images.githubusercontent.com/41000648/163947218-241abd3f-e978-40e0-8dd5-e01cda49ff56.png">

---


- If Site title is lengthy then short name is displayed on phone's Add to home screen 
![20220419_124918](https://user-images.githubusercontent.com/41000648/163947740-84da8a88-d87f-49d2-abec-a5046ffc106e.jpg)

---

- If Short Name filter is added via plugin then the field appears disabled
<img width="723" alt="Screenshot 2022-04-19 at 12 37 45 PM" src="https://user-images.githubusercontent.com/41000648/163948458-1bccd165-5856-40d5-9965-cf4853b8a263.png">

---

- If Shortname is added then site health screen shows it as passed
<img width="978" alt="Screenshot 2022-04-19 at 12 59 06 PM" src="https://user-images.githubusercontent.com/41000648/163949252-71d194a1-f32a-4777-9681-14460afb407e.png">

---

- If short name is not added then site health shows error and a link general settings to add short name

<img width="1036" alt="Screenshot 2022-04-19 at 12 58 24 PM" src="https://user-images.githubusercontent.com/41000648/163949125-ca0059bd-6249-473a-9544-0df1dc41f333.png">

---

**Comment by morsssss** [comment]
+1 to adding this new field. My WP experience is limited, but I'm helping a local news site called "Montclair Local" to make some fixes to their brand new site. The plugin shortened the name to "MntclrLclNws". I realized this plugin was generating the web app manifest, I found and installed [Weston's helpful gist](https://gist.github.com/westonruter/dec7d190060732e29a09751ab99cc549), and I fixed the name. Many others would never figure out how to do this.
